vizro-core/docs/pages/user-guides/extensions.md

```diff
@@ -88,3 +88,111 @@ For more information, see our documentation on [customizing CSS](custom-css.md)
+            components:
+                - type: "graph"
+                    figure:
+                        _target_: "__main__.custom_bar"
+                        data_frame: "iris"
+                        x: "sepal_length"
+                        y: "sepal_width"
+    """
+
+    # Load dashboard from YAML
+    dashboard_config = yaml.safe_load(dashboard_yaml)# (2)!
+    dashboard = vm.Dashboard(**dashboard_config)
+
+    # Build and run the dashboard
+    app = Vizro().build(dashboard)
+
+    if __name__ == "__main__":
+        app.run()
+    ```
+
+    1. Define the custom chart either in `app.py` or in a separate file.
+    1. Parse the YAML or JSON configuration that refers to the custom chart with the **correct import path**. If the custom chart is defined in the `app.py`, then use `__main__` as the import path.
+
+### Validating dashboards without executing `CapturedCallable` functions
+
+It is possible to validate a dashboard configuration without executing the `CapturedCallable` functions.
+
+You can use this method when you want to check if the dashboard configuration is valid, but you don't want to execute the custom functions until run-time, which may be in a sandboxed environment. This is useful when the custom functions are not available at validation time, or when they originate from untrusted sources (e.g. when a large language model is used to generate that code).
+
+!!! note When starting the server
+
+    If you validate your dashboard configuration with a `CapturedCallable` function that is undefined, you will not be able to start the server. Hence when running the dashboard, you will need to recreate a dashboard object with all the `CapturedCallable` functions defined.
+
+!!! example "Validating dashboards without executing `CapturedCallable` functions"
+
+    ```python
+    import vizro.models as vm
+    import vizro.plotly.express as px
+    from vizro import Vizro
+    from vizro.managers import data_manager
+    from vizro.models.types import AllowedCapturedCallable
+
+    data_manager["iris"] = px.data.iris()
+
+    dashboard_config = {
+        "title": "Test dashboard",
+        "pages": [
+            {
+                "title": "Page 1",
+                "components": [
+                    {
+                        "type": "ag_grid",
+                        "figure": {"_target_": "llm_generated_grid", "data_frame": "iris"},
+                    },
+                ],
+            }
+        ],
+    }
+
+    dashboard = vm.Dashboard.model_validate(
+        dashboard_config,
+        context={
+            "allowed_undefined_captured_callables": [
+                AllowedCapturedCallable(function_name="llm_generated_grid", mode="ag_grid")
+            ]
+        },
+    )
+    app = Vizro().build(dashboard)  # (1)!
+    ```
+
+    1. Because this dashboard configuration contains a `CapturedCallable` function that is undefined, you will not be able to start the server. Hence `app.run()` will raise an error.
```

Comment:
Something like this might be clearer? Assuming I understood the behaviour correctly.

```suggestion
    app = Vizro().build(dashboard)  # (1)!
    try:
        app.run()  # (2)!
    except Exception as exc:
        print(exc)
    ```

    1. The dashboard configuration contains a `CapturedCallable` function that is undefined but allowed by `allowed_undefined_captured_callables`. The app can still be built without raising any errors.
    2. However, it is not possible to run the app without undefined `CapturedCallable`s. This raises an error.
```